controllers/clusters/cadence_controller.go

```diff
@@ -458,232 +440,396 @@ func (r *CadenceReconciler) handleDeleteCluster(
-	providerAccountName := c.Spec.DataCentres[0].ProviderAccountName
-
-	cassandraDataCentres := []*v1beta1.CassandraDataCentre{
-		{
-			GenericDataCentreSpec: v1beta1.GenericDataCentreSpec{
-				Name:                dcName,
-				Region:              dcRegion,
-				CloudProvider:       cloudProvider,
-				ProviderAccountName: providerAccountName,
-				Network:             network,
-			},
-			NodeSize:                       cassNodeSize,
-			NodesNumber:                    cassNodesNumber,
-			ReplicationFactor:              cassReplicationFactor,
-			PrivateIPBroadcastForDiscovery: cassPrivateIPBroadcastForDiscovery,
-		},
+	switch app {
+	case models.CassandraAppKind:
+		genericClusterSpec.Name = models.CassandraChildPrefix + c.Name
+		spec := v1beta1.CassandraSpec{
+			GenericClusterSpec:  genericClusterSpec,
+			PasswordAndUserAuth: true,
+			PCICompliance:       c.Spec.PCICompliance,
+			BundledUseOnly:      true,
+		}
+		return spec
+
+	case models.KafkaAppKind:
+		genericClusterSpec.Name = models.KafkaChildPrefix + c.Name
+		spec := v1beta1.KafkaSpec{
+			GenericClusterSpec:        genericClusterSpec,
+			PCICompliance:             c.Spec.PCICompliance,
+			BundledUseOnly:            true,
+			ReplicationFactor:         3,
+			PartitionsNumber:          3,
+			AllowDeleteTopics:         true,
+			AutoCreateTopics:          true,
+			ClientToClusterEncryption: c.Spec.DataCentres[0].ClientEncryption,
+		}
+
+		return spec
+
+	case models.OpenSearchAppKind:
+		managerNodes := []*v1beta1.ClusterManagerNodes{{
+			NodeSize: c.Spec.CalculateNodeSize(
+				c.Spec.DataCentres[0].CloudProvider,
+				c.Spec.PackagedProvisioning[0].SolutionSize,
+				models.OpenSearchAppKind,
+			),
+			DedicatedManager: false,
+		}}
+
+		genericClusterSpec.Name = models.OpenSearchChildPrefix + c.Name
+		spec := v1beta1.OpenSearchSpec{
+			GenericClusterSpec:  genericClusterSpec,
+			PCICompliance:       c.Spec.PCICompliance,
+			BundledUseOnly:      true,
+			ClusterManagerNodes: managerNodes,
+		}
+
+		return spec
 	}
 
-	spec := v1beta1.CassandraSpec{
-		GenericClusterSpec: v1beta1.GenericClusterSpec{
-			Name:            models.CassandraChildPrefix + c.Name,
-			Version:         latestCassandraVersion,
-			SLATier:         slaTier,
-			PrivateNetwork:  privateNetwork,
-			TwoFactorDelete: twoFactorDelete,
-		},
-		DataCentres:         cassandraDataCentres,
-		PasswordAndUserAuth: cassPasswordAndUserAuth,
-		PCICompliance:       pciCompliance,
-		BundledUseOnly:      true,
+	return nil
+}
+
+func (r *CadenceReconciler) reconcileNetwork(availableCIDRs []string, app string) (string, error) {
+	switch app {
+	case models.CassandraAppKind:
+		return availableCIDRs[1], nil
+	case models.KafkaAppKind:
+		return availableCIDRs[2], nil
+	case models.OpenSearchAppKind:
+		return availableCIDRs[3], nil
 	}
 
-	return &v1beta1.Cassandra{
-		TypeMeta:   typeMeta,
-		ObjectMeta: metadata,
-		Spec:       spec,
-	}, nil
+	return "", nil
+}
```

Comment:
It seems that errors are not being returned in this function. Perhaps consider removing the error from the return parameters